### PR TITLE
fileshare: init at 0.2.4

### PR DESCRIPTION
**Pull request by esclear**
###### Motivation for this change
Fileshare is a simple tool for sharing the contents of a directory via a
webserver and optionally allowing uploads.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
  (No dependencies)
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
  (+ 44.51 MB)
- [x] Ensured that relevant documentation is up to date
  (No documentation)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers
CC: myself :)

### CODE REVIEW

**Comment by tilpner** [comment]
@GrahamcOfBorg build fileshare

File: pkgs/servers/fileshare/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  
+  postPatch = ''
+    sed -i 's,$(shell git rev-parse --short HEAD),/${version},g' Makefile
+  '';
+
+  nativeBuildInputs = [ pkgconfig git ];
+  buildInputs = [ libmicrohttpd ];
+  
+  buildPhase = ''
+    make BUILD=release
```
Comment (teto) [inline pkgs/servers/fileshare/default.nix]: better pass this as a makeFlags else you prevent someone from using custom hooks.

Comment (esclear) [reply pkgs/servers/fileshare/default.nix]: I was sure that this kind of flag existed, but was unable to find it in the documentation.
Thanks, I've done so.

File: pkgs/servers/fileshare/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  makeFlags = "BUILD=release";
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp bin/release/fileshare $out/bin
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A small HTTP Server for quickly sharing files over the network.";
```
Comment (mmahut) [inline pkgs/servers/fileshare/default.nix]: No dot at the end please.

Comment (esclear) [reply pkgs/servers/fileshare/default.nix]: Fixed

**Comment by mmahut** [comment]
@GrahamcOfBorg build fileshare

**Comment by teto** [comment]
it would be nice if upstream's Makefile would allow VCSVERSION / CC overrides. Meanwhile this works. (of the same type, there is webfs already packaged in nixpkgs).

**Pull request merged**